utils/helpers.py

```diff
@@ -324,3 +328,49 @@ def decode_error_output(data_hex):
         return f"Panic(uint256): {desc}"
     # 4) Unknow format
     return f"Unknown revert payload: {data_hex}"
+
+
+def withdraw_neon_to_solana_eth_sign(web3_client, withdraw_from, withdraw_to, withdraw_contract):
+    amount = web3_client.get_balance(withdraw_from)
+    tx = web3_client.make_raw_tx(from_=withdraw_from, amount=amount)
+    """
+        withdraw contract requires trx value to be divisible to 10**9
+        remainings of the value are dropped with // operation
+    """
+    value = (amount - web3_client.eth.estimate_gas(tx) * web3_client.gas_price()) // 10**9
+    tx["value"] = value * 10**9
+    instruction_tx = withdraw_contract.functions.withdraw_on_chain(bytes(withdraw_to.pubkey())).build_transaction(tx)
+    receipt = web3_client.send_transaction(withdraw_from, instruction_tx)
+    assert receipt["status"] == 1
+
+
+def withdraw_neon_to_solana_sol_sign(
+    withdraw_from, withdraw_to, withdraw_contract, evm_loader, web3_client_sol, treasury_pool
+):
+    ata = evm_loader.create_associate_token_acc(withdraw_to, withdraw_to, WRAPPED_SOL_MINT)
+    spl_token = SplToken(evm_loader, WRAPPED_SOL_MINT, TOKEN_PROGRAM_ID, withdraw_to)
+    ata_balance_before = int(spl_token.get_balance(ata, commitment=Confirmed).value.amount)
+
+    amount = web3_client_sol.get_balance(withdraw_from.checksum_address)
+    data = decode_function_signature("withdraw_on_chain(bytes32)", [bytes(withdraw_to.pubkey())])
+    trx_estimate_obj = ScheduledTrxEstimateRequest(
+        withdraw_from.checksum_address, withdraw_contract.address, data, amount
+    )
+    estimate_result = web3_client_sol.estimate_scheduled(withdraw_from.solana_account.pubkey(), [trx_estimate_obj])
+    gas = web3_client_sol.gas_price() * int(estimate_result["gasList"][0], 16)
+    """
+        withdraw contract requires trx value to be divisible to 10**9
+        remainings of the value are dropped with // operation
+    """
+    trx_estimate_obj.value = ((trx_estimate_obj.value - gas) // 10**9) * 10**9
+
+    if trx_estimate_obj.value > 0:
```

Comment:
strange condition
you can move it up, check amount and no to do estimation for the case as well